modules/standard/OS.chpl

```diff
@@ -1298,53 +1328,135 @@ module OS {
   }
 
   /*
-     :class:`IoError` is the subclass of :class:`SystemError` that serves as the
-     base class for all errors regarding inputs and their formatting.
-     They are typically not directly generated by the OS, but they are
-     used and emitted by the IO module.
+     :class:`IoError` is the subclass of :class:`SystemError` corresponding to
+     EIO.
   */
   class IoError : SystemError {
-    proc init(err: errorCode, details: string = "") {
+    proc init(err: errorCode = EIO, details: string = "") {
       super.init(err, details);
     }
   }
 
   /*
-     :class:`EofError` is the subclass of :class:`IoError` corresponding to
+     :class:`EofError` is the the Chapel-specific error corresponding to
      encountering end-of-file.
   */
-  class EofError : IoError {
-    proc init(details: string = "", err: errorCode = EEOF:errorCode) {
-      super.init(err, details);
+  class EofError : Error {
+    var details: string;
+    var err_msg: string;
```

Comment:
Not imperative, but you could use the `_msg` field from `Error` for one of these strings to avoid the extra allocation.